internal/campaigns/published_test.go

```diff
@@ -0,0 +1,82 @@
+package campaigns
+
+import (
+	"encoding/json"
+	"testing"
+)
+
+func TestPublishedValue(t *testing.T) {
+	tests := []struct {
+		name  string
+		val   interface{}
+		True  bool
+		False bool
+		Draft bool
+	}{
+		{name: "True", val: true, True: true, False: false, Draft: false},
+		{name: "False", val: false, True: false, False: true, Draft: false},
+		{name: "Draft", val: "draft", True: false, False: false, Draft: true},
+	}
```

Comment:
I'd add a sanity check:

```suggestion
            {name: "Invalid", val: 18, Invalid: true}
	}
```